easybuild/framework/easyblock.py

```diff
@@ -1428,6 +1428,27 @@ def make_module_extra(self, altroot=None, altversion=None):
                                      value, type(value))
             lines.append(self.module_generator.append_paths(key, value, allow_abs=self.cfg['allow_append_abs_path']))
 
+        # Add automatic PYTHONPATH or EBPYTHONPREFIXES if they aren't already present and python paths exist
+        if not os.path.isfile(f'{self.installdir}/bin/python'):  # only needed when not a base python installation
+            python_paths = [os.path.relpath(path, self.installdir)
+                            for path in glob.glob(f'{self.installdir}/lib/python*/site-packages')
+                            if re.match(self.installdir + r'/lib/python\d+\.\d+/site-packages', path)]
```

Comment:
I'd be careful with using user-supplied input in a regex. It could happen to be an invalid regex. Maybe:
```suggestion
                            if re.search(r'/lib/python\d+\.\d+/site-packages$', path)]
```

Reply:
Hm, i think this would trip up on a few corner cases though, like i know fsl package does this thing
```
                      extra fsl subdir here
                          | 
                          v 
/apps/software/fsl/1.2.3/fsl/lib/python3.11/site-packages/
```

I could do 
```suggestion
                            if re.match(r'lib/python\d+\.\d+/site-packages$', os.path.relpath(self.installdir, path))]
```

Reply:
For `/apps/software/fsl/1.2.3/fsl/lib/python3.11/site-packages/` the glob `{self.installdir}/lib/python*/site-packages'` won't match will it? So really all we need to do in the regexp is to verify the `python*` folder name which we can't do in the glob directly.

But sure this would also work. You can omit the trailing $ when using `match`. Maybe even call relpath only once:
```
[relpath
 for relpath in (os.path.relpath(path, self.installdir) for path in glob.glob(f'{self.installdir}/lib/python*/site-packages'))
 if re.match(r'/lib/python\d+\.\d+/site-packages', relpath)]
```